pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -355,9 +390,41 @@ def build_model(
                 dims="date",
             )
 
-            intercept = self.intercept_dist(
-                name="intercept", **self.model_config["intercept"]["kwargs"]
-            )
+            if self.time_varying_intercept:
```

Comment:
Can we pull this all out into a function, say, `create_time_varying_intercept`. 
Think it'd be much easier to follow as a whole

```python
if self.time_varying_intercept: 
    intercept = create_time_varying_intercept(...)
    # or 
    intercept = pm.Deterministic(
         name="intercept", 
         var=create_time_varying_intercept(...), 
         dims="date",  
    )
else: 
    intercept = self.intercept_dist(
        name="intercept", 
        **self.model_config["intercept"]["kwargs"]
    )

Reply:
This is neat, but has to be slightly less concise and elegant to accommodate for other time varying coefficients (like media channels).

Reply:
✅